src/contracts/user-deployer-whitelist.ts

```diff
@@ -92,4 +116,48 @@ export class UserDeployerWhitelist extends Contract {
       address: Address.UnmarshalPB(deployerContract.getContractAddress()!)
     }))
   }
+
+  /**
+   * @param TierID tierId.
+   * @returns Tier object containing TierDetails.
+   */
+  async getTierInfoAsync(tierId: TierID): Promise<ITier> {
+    const req = new GetTierInfoRequest()
+    req.setId(tierId)
+    const result = await this.staticCallAsync('GetTierInfo', req, new GetTierInfoResponse())
+    return {
+      tierId: result.getTier()!.getTierId(),
+      fee: unmarshalBigUIntPB(result.getTier()!.getFee()!),
+      name: result.getTier()!.getName()
+    }
+  }
+
+  /**
+   * Allows to modify TierInfo, caller can only be owner set in init.
+   *
+   * @param TierID tierId
+   * @param fee
+   * @param name
+   */
+
+  async setTierInfoAsync(tierId: TierID, fee?: BN, name?: string): Promise<void> {
```

Comment:
This method should take `ITier` as a non-optional parameter, and then fill in the request from the passed in object. It shouldn't be calling getTierInfoAsync.

Reply:
OK, but in loomchain cli, we decided to keep fee and name optional and TIerID as mandatory, should  not the protocol work same here, if we won't call  getTierInfoAsync, then how will we populate fields which don't need modification.

Reply:
This isn't a CLI, this is an API